dotnet/src/InternalUtilities/src/Data/VectorStoreRecordPropertyReader.cs

```diff
@@ -15,73 +14,385 @@ namespace Microsoft.SemanticKernel.Data;
+            return this._dataPropertiesInfo!;
+        }
+    }
+
+    /// <summary>Gets the vector property info objects.</summary>
+    public IReadOnlyList<PropertyInfo> VectorPropertiesInfo
+    {
+        get
+        {
+            this.LoadPropertyInfoIfNeeded();
+            return this._vectorPropertiesInfo!;
+        }
+    }
+
+    /// <summary>Gets the name of the first vector property in the definition or null if there are no vectors.</summary>
+    public string? FirstVectorPropertyName => this._vectorProperties.FirstOrDefault()?.DataModelPropertyName;
+
+    /// <summary>Gets the first vector PropertyInfo object in the data model or null if there are no vectors.</summary>
+    public PropertyInfo? FirstVectorPropertyInfo => this.VectorPropertiesInfo.Count > 0 ? this.VectorPropertiesInfo[0] : null;
+
+    /// <summary>Gets the property name of the first key property in the definition.</summary>
+    public string KeyPropertyName => this._keyProperties[0].DataModelPropertyName;
+
+    /// <summary>Gets the storage name of the first key property in the definition.</summary>
+    public string KeyPropertyStoragePropertyName => this._keyPropertyStoragePropertyNames.Value[0];
+
+    /// <summary>Gets the storage names of all the properties in the definition.</summary>
+    public IReadOnlyDictionary<string, string> StoragePropertyNamesMap => this._storagePropertyNamesMap.Value;
+
+    /// <summary>Gets the storage names of the key properties in the definition.</summary>
+    public IReadOnlyList<string> KeyPropertyStoragePropertyNames => this._keyPropertyStoragePropertyNames.Value;
+
+    /// <summary>Gets the storage names of the data properties in the definition.</summary>
+    public IReadOnlyList<string> DataPropertyStoragePropertyNames => this._dataPropertyStoragePropertyNames.Value;
+
+    /// <summary>Gets the storage names of the vector properties in the definition.</summary>
+    public IReadOnlyList<string> VectorPropertyStoragePropertyNames => this._vectorPropertyStoragePropertyNames.Value;
+
+    /// <summary>Gets the json name of the first key property in the definition.</summary>
+    public string KeyPropertyJsonName => this.KeyPropertyJsonNames[0];
+
+    /// <summary>Gets the json names of the key properties in the definition.</summary>
+    public IReadOnlyList<string> KeyPropertyJsonNames => this._keyPropertyJsonNames.Value;
+
+    /// <summary>Gets the json names of the data properties in the definition.</summary>
+    public IReadOnlyList<string> DataPropertyJsonNames => this._dataPropertyJsonNames.Value;
+
+    /// <summary>Gets the json names of the vector properties in the definition.</summary>
+    public IReadOnlyList<string> VectorPropertyJsonNames => this._vectorPropertyJsonNames.Value;
+
+    /// <summary>A map of data model property names to the names they will have if serialized to JSON.</summary>
+    public IReadOnlyDictionary<string, string> JsonPropertyNamesMap => this._jsonPropertyNamesMap.Value;
```

Comment:
Yeah, you are right.  It would make sense to split this into a subclass.  Let's get this checked in first, and then if there is time to refactor further that would make sense to add.